kie-wb-common-react/boxed-expression-editor/src/components/ExpressionContainer/ExpressionContainer.tsx

```diff
@@ -28,43 +28,53 @@ import {
   SimpleListItemProps,
 } from "@patternfly/react-core";
 import { PopoverMenu } from "../PopoverMenu";
+import { ExpressionProps, LiteralExpressionProps, LogicType } from "../../api";
+import { LiteralExpression } from "../LiteralExpression";
 
 export interface ExpressionContainerProps {
-  /** The name of the expression */
-  name: string;
-  /** The type of the expression */
-  type?: string;
-  /** Selected expression is already present */
-  selectedExpression?: string;
+  /** Expression properties */
+  selectedExpression: ExpressionProps;
 }
 
-export const ExpressionContainer: (props: ExpressionContainerProps) => JSX.Element = (
+export const ExpressionContainer: ({ selectedExpression }: ExpressionContainerProps) => JSX.Element = (
   props: ExpressionContainerProps
 ) => {
   const { i18n } = useBoxedExpressionEditorI18n();
 
-  const [logicTypeIsPresent, setLogicTypeSelected] = useState(!_.isEmpty(props.selectedExpression));
+  const [logicTypeIsPresent, setLogicTypeSelected] = useState(
+    !_.isEmpty(props.selectedExpression.logicType) || props.selectedExpression.logicType === LogicType.Undefined
+  );
   const [actionDropdownIsOpen, setActionDropDownOpen] = useState(false);
-  const [selectedExpression, setSelectedExpression] = useState(props.selectedExpression || i18n.selectExpression);
+  const [selectedExpression, setSelectedExpression] = useState(props.selectedExpression);
 
   const onLogicTypeSelect = useCallback(
     (currentItem: React.RefObject<HTMLButtonElement>, currentItemProps: SimpleListItemProps) => {
       setLogicTypeSelected(true);
-      setSelectedExpression(currentItemProps.children as string);
+      const selectedLogicType = currentItemProps.children as LogicType;
+      setSelectedExpression({
+        ...selectedExpression,
+        logicType: selectedLogicType,
+      });
     },
-    []
+    [selectedExpression]
   );
 
   const executeClearAction = useCallback(() => {
     setLogicTypeSelected(false);
-    setSelectedExpression(i18n.selectExpression);
-  }, [i18n.selectExpression]);
+    setSelectedExpression({
+      name: selectedExpression.name,
+      dataType: selectedExpression.dataType,
+      logicType: LogicType.Undefined,
+    });
+  }, [selectedExpression]);
 
-  const renderExpressionActionsDropdown = () => {
+  const renderExpressionActionsDropdown = useCallback(() => {
     return (
       <Dropdown
         onSelect={() => setActionDropDownOpen(!actionDropdownIsOpen)}
-        toggle={<KebabToggle onToggle={(isOpen) => setActionDropDownOpen(isOpen)} id="expression-actions-toggle" />}
+        toggle={
+          <KebabToggle onToggle={(isOpen) => setActionDropDownOpen(isOpen)} className="expression-actions-toggle" />
```

Comment:
Creating an arrow function inline like this will trigger a re-render on KebabToggle even when nothing has changed, since React compares props by reference. Every time you create a new arrow function, React thinks something has changes on this component's props, so it has to re-evaluate everything. You can wrap this function inside a "useCallback" to avoid such behaviour. While this might seem irrelevant now and React is very fast on re-rendering things, this may present as a potential bug in the future, caused by this unnecessary re-rendering.